### PR TITLE
feat(combobox): Convert to functional component and add translation function for status text

### PR DESCRIPTION
**Pull request by vibdev**
## Summary

To support extending the combobox to support more features I ported the core component to a React Functional Component. ~Nothing in behavior should be changed.~ JK updated a prop name and added a new prop. Let me know if those should go in a separate PR

## Checklist

- [x] branch has been rebased on the latest master commit
- [x] tests are changed or added
- [x] `yarn test` passes
- [x] all (dev)dependencies that the module needs is added to its `package.json`
- [x] code adheres to the [API & Pattern guidelines](../API_PATTERN_GUIDELINES.md)

### CODE REVIEW

File: modules/_labs/combobox/react/lib/Combobox.tsx [hunk 1]
```diff
@@ -121,54 +118,85 @@ const listBoxIdPart = `listbox`;
...
+  return `There ${listCount === 1 ? 'is' : 'are'} ${listCount} suggestion${
+    listCount === 1 ? '' : 's'
+  }.`;
+};
+
+const isValidSingleChild = (child: React.ReactNode) => {
+  return React.isValidElement(child) && React.Children.only(child);
+};
+
+const Combobox: React.FC<ComboboxProps> = ({
```
Comment (anicholls) [inline modules/_labs/combobox/react/lib/Combobox.tsx]: Nit: We've been recommending using `const Component = (props: ComponentProps) => {}` over this format for a reason I can't seem to think of at the moment haha. I will look into it and get back to you, but do you mind updating the formatting here so it aligns with our other functional components?

Comment (vibdev) [reply modules/_labs/combobox/react/lib/Combobox.tsx]: Looks like it uncovered a potential bug by typing children.props better

Comment (vibdev) [reply modules/_labs/combobox/react/lib/Combobox.tsx]:
```
const Component: React.FC<ComponentProps> = (props) => {
  props.children.props.helicopter // helicopter is typed as any
}
```
```
const Component = (props: ComponentProps) => {
    props.children.props.helicopter // helicopter is typed correctly as Badass
}
```

Comment (anicholls) [reply modules/_labs/combobox/react/lib/Combobox.tsx]: That was why! Thanks for testing it out 👍

**Pull request merged**